src/pages/confirm-test-details/confirm-test-details.page.ts

```diff
@@ -0,0 +1,249 @@
+          select(getCandidate),
+          select(getUntitledCandidateName),
+        ),
+        candidateName$: currentTest$.pipe(
+          select(getJournalData),
+          select(getCandidate),
+          select(getCandidateName),
+        ),
+        startDateTime$: currentTest$.pipe(
+          select(getJournalData),
+          select(getTestSlotAttributes),
+          select(getTestStartDateTime),
+        ),
+        testOutcomeText$: currentTest$.pipe(
+          select(getTestOutcomeText),
+        ),
+        activityCode$: currentTest$.pipe(
+          select(getActivityCode),
+        ),
+        testCategory$: currentTest$.pipe(
+          select(getTestCategory),
+          map(testCategory => testCategory as TestCategory),
+        ),
+        provisionalLicense$: currentTest$.pipe(
+          select(getPassCompletion),
+          map(isProvisionalLicenseProvided),
+        ),
+        transmission$: currentTest$.pipe(
+          select(vehicleDetails.vehicleDetails),
+          select(getGearboxCategory),
+        ),
+        d255$: currentTest$.pipe(
+          select(getTestSummary),
+          select(getD255),
+        ),
+      };
+    });
+
+    const { testCategory$, testOutcomeText$, candidateUntitledName$ } = this.pageState;
+
+    this.merged$ = merge(
+      testCategory$.pipe(map(value => this.category = value)),
+      testOutcomeText$.pipe(map(value => this.testOutcome = value)),
+      candidateUntitledName$.pipe(tap(value => this.candidateName = value)),
+    );
+  }
+
+  isTerminated(testResult: string): boolean {
+    return testResult === TestOutcome.Terminated;
+  }
+
+  isPassed(testResult: string): boolean {
+    return testResult === TestOutcome.Passed;
+  }
+
+  getActivityCode(activityCodeModel: ActivityCodeModel): ActivityCode {
+    return activityCodeModel.activityCode;
+  }
+
+  getProvisionalText(received: boolean): LicenceReceivedText {
+    return received ? LicenceReceivedText.TRUE : LicenceReceivedText.FALSE;
+  }
+
+  getTransmissionText(gearbox: GearboxCategory): GearBox {
+    return gearbox === GearBox.MANUAL ? GearBox.MANUAL : GearBox.AUTOMATIC;
+  }
+
+  getD255Text(d255: boolean): D255 {
+    return d255 ? D255.TRUE : D255.FALSE;
+  }
+
+  async onSubmit() {
+    await this.showConfirmTestDetailsModal();
+  }
+
+  async showConfirmTestDetailsModal() {
+    const alert = this.alertController.create({
+      message: `You are about to submit a Cat ${this.category} ${this.testOutcome} for ${this.candidateName}
+                <br/><br/>Are you sure you want to submit this result?`,
+      cssClass: 'confirm-declaration-modal',
+      buttons: [
+        {
+          text: 'Cancel',
+          handler: () => {
+          },
+        },
+        {
+          text: 'Submit',
+          handler: () => this.persistAndNavigate(),
+        },
+      ],
+      enableBackdropDismiss: false,
+    });
+    await alert.present();
+  }
+
+  async persistAndNavigate() {
```

Comment:
again, would be good to have a more descriptive name, e.g. `onTestDetailsConfirm()` plus add return type `Promise<void>`

Reply:
was carryed over from health dec screen, but happy to rename. and your right so ive added a return type to this and showConfirmTestDetailsModal